### ISSUE TITLE
UnhandledPromiseRejection on this.context.authRequest

### ISSUE THREAD

**Issue by pooshonbanerjee** [MichaelJamesParsons/ustream-nodejs-sdk#5]
For ftp.put there are a series of actions bound by promises. The first of which is getting the ftp details from the UStream API, this needs a GET call with correct Bearer Token as shown below:

```
  authRequest (verb, route, data, headers) {
    return new Promise((resolve, reject) => {
      this.authorize(false)
        .then(() => {
          this.request(verb, route, data, this._buildAuthHeaders(headers)).then((res) => {
            return resolve(res)
          })
        })
        .catch((err) => {
          reject(err)
        })
    })
  }

```

If the token has expired/bad/missing the GET returns Unauthorized, this rejects the Promise, which is not handled as show below.

```
    return this._initiateUpload(channelId, opts)
      .then((res) => {
        return self._ftpUpload(res.host, res.user, res.password, res.port, `${res.path}.${ext}`, file.stream, file.fpath)
          .then(() => {
            return self._completeUpload(channelId, res['videoId'], 'ready')
          })
      })

```
It might be a good idea to add a catch block to each .then() here.

**Comment by MichaelJamesParsons** [comment]
I apologize for the delay. I was out sick for a few days.

> It might be a good idea to add a catch block to each .then() here.

This won't be necessary. When a function returns a chain of promises, JavaScript will "follow" the chain until all promises have resolved. Any errors that occur along the chain will be handled by your `catch` statement.

The bug resides in the call to `this.request()` in this block of code:

```javascript
return new Promise((resolve, reject) => {
    this.authorize(false)
        .then(() => {
          this.request(verb, route, data, this._buildAuthHeaders(headers)).then((res) => {
            return resolve(res)
          })
        })
        ...
```
The code returns a `Promise`, then waits for multiple unchained promises to complete before resolving. Because the function does not return a chain of promises, any errors that occur in `this.request()` will not be caught. A simple refactor will allow errors to be passed directly to your own `catch` block, instead of being handled internally.

```javascript
return this.authorize(false).then(() => {
  return this.request(verb, route, data, this._buildAuthHeaders(headers))
})
```

**Comment by MichaelJamesParsons** [comment]
A fix for this has been released in `1.2.2`.
